src/libraries/NormalizedAmount.sol

```diff
@@ -132,16 +132,18 @@ library NormalizedAmountLib {
         return a.amount < b.amount ? a : b;
     }
 
-    /// @dev scale the amount from origDecimals to normDecimals (base 10)
+    /// @dev scale the amount from original decimals to target decimals (base 10)
     function scale(
         uint256 amount,
         uint8 fromDecimals,
         uint8 toDecimals
     ) internal pure returns (uint256) {
         if (fromDecimals > toDecimals) {
             return amount / (10 ** (fromDecimals - toDecimals));
-        } else {
+        } else if (fromDecimals < toDecimals) {
             return amount * (10 ** (toDecimals - fromDecimals));
+        } else {
+            return amount;
         }
     }
```

Comment:
it may be a good idea to keep this logic consistent with the solana impl.
https://github.com/wormholelabs-xyz/solana-example-native-token-transfers/blob/ab2ca08b3dd27881a82070ad0e09964aba73d590/solana/programs/example-native-token-transfers/src/normalized_amount.rs#L47

Reply:
The structure is identical -- does re-ordering have a significant impact on gas or something else?

Reply:
its more readable.